shipyard_blueprints/dropbox/shipyard_dropbox/cli/upload.py

```diff
@@ -193,64 +118,80 @@ def get_dropbox_client(access_key):
-    destination_folder_name = clean_folder_name(args.destination_folder_name)
-    if not os.path.exists(destination_folder_name) and (destination_folder_name != ""):
-        os.makedirs(destination_folder_name)
-
-    client = get_dropbox_client(access_key=access_key)
-
-    if source_file_name_match_type == "regex_match":
-        file_names = find_dropbox_file_names(client=client, prefix=source_folder_name)
-        matching_file_names = find_matching_files(
-            file_names, re.compile(source_file_name)
+    try:
+        args = get_args()
+        access_key = args.access_key
+        source_file_name = args.source_file_name
+        source_folder_name = args.source_folder_name
+        source_full_path = shipyard.combine_folder_and_file_name(
+            folder_name=os.path.join(os.getcwd(), source_folder_name),
+            file_name=source_file_name,
+        )
+        destination_folder_name = shipyard.clean_folder_name(
+            args.destination_folder_name
         )
-        print(f"{len(matching_file_names)} files found. Preparing to download...")
 
-        for index, file_name in enumerate(matching_file_names):
-            destination_name = determine_destination_name(
+        client = get_dropbox_client(access_key=access_key)
+        if args.source_file_name_match_type == "exact_match":
+            destination_full_path = shipyard.determine_destination_full_path(
                 destination_folder_name=destination_folder_name,
                 destination_file_name=args.destination_file_name,
-                source_full_path=file_name,
-                file_number=index + 1,
+                source_full_path=source_full_path,
             )
-
-            if not file_name.startswith("/"):
-                file_name = f"/{file_name}"
-            print(f"Downloading file {index+1} of {len(matching_file_names)}")
-            download_dropbox_file(
-                file_name=file_name,
+            upload_dropbox_file(
+                source_full_path=source_full_path,
+                destination_full_path=destination_full_path,
                 client=client,
-                destination_file_name=destination_name,
             )
-    else:
-        destination_name = determine_destination_name(
-            destination_folder_name=destination_folder_name,
-            destination_file_name=args.destination_file_name,
-            source_full_path=source_full_path,
-        )
 
-        if not source_full_path.startswith("/"):
-            source_full_path = f"/{source_full_path}"
-        download_dropbox_file(
-            file_name=source_full_path,
-            client=client,
-            destination_file_name=destination_name,
-        )
+        elif args.source_file_name_match_type == "regex_match":
+            file_names = shipyard.find_all_local_file_names(source_folder_name)
+            matching_file_names = shipyard.find_all_file_matches(
+                file_names, re.compile(source_file_name)
```

Comment:
For consistency, I would have done the:

```python
if n_matches := len(matching_file_names) == 0:
    raise ExitCodeException...

logger.info(f"{n_matches} found. Preparing to upload...")
```

Reply:
Not a huge deal though

Reply:
I actually opted out of this intentionally:
`if n_matches := len(matching_file_names) == 0:`
ends up setting n_matches to a bool value
which ends up logging `TRUE found. Preparing to upload...`

Reply:
yes I had that happen to me too, the fix is to wrap it around () [](edited-with-Axolo)

Reply:
`if (n_matches := len(matching_file_names)) == 0` [](edited-with-Axolo)

Reply:
it is a stylistic choice at that point. Not a fan of the `(`
I think
`number_of_matches= len(matching_file_names)`
`if number_of_matches == 0:`
is more readable  but in this specific case since `number_of_matches`  would only be used twice `len()` seemed appropriate [](edited-with-Axolo)